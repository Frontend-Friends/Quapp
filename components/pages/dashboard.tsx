import React, { FC, useState } from 'react'
import { CondensedContainer } from '../condensed-container'
import { Header } from '../header'
import { Box, Fab, Grid, IconButton, Modal, Typography } from '@mui/material'
import SpaceItem from '../spaces/space-item'
import AddIcon from '@mui/icons-material/Add'
import SpaceForm from '../../pages/community/space-form'
import { SpaceItemTypeWithUser } from '../products/types'
import { UseTranslationType } from '../../hooks/use-translation'
import EditSpaceModal from '../spaces/edit-space-modal'
import { useSnackbar } from '../../hooks/use-snackbar'
import { User } from '../user/types'
import CloseIcon from '@mui/icons-material/Close'

export const Dashboard: FC<{
  spaces?: SpaceItemTypeWithUser[]
  user: User
  t: UseTranslationType
}> = ({ spaces, user, t }) => {
  const [isLoading, setIsLoading] = useState(false)
  const [mySpaces, setMySpaces] = useState<SpaceItemTypeWithUser[]>(
    spaces ?? []
  )
  const [openModal, setOpenModal] = useState<boolean>(false)
  const [openEditModal, setOpenEditModal] = useState<boolean>(false)
  const [space, setSpace] = useState<SpaceItemTypeWithUser | null>(null)
  const setAlert = useSnackbar((state) => state.setAlert)
  return (
    <CondensedContainer className="relative">
      <Header title={t('SPACES_title')} />
      {!!mySpaces?.length ? (
        <Grid container columns={{ md: 1 }} spacing={{ xs: 4 }} pt={4}>
          {mySpaces.map((mySpace) => {
            const isOwner = mySpace.adminId === user.id

            return (
              <SpaceItem
                key={`${mySpace.creationDate}-${mySpace.id}`}
                space={mySpace}
                setMySpaces={setMySpaces}
                mySpaces={mySpaces}
                setOpenEditModal={setOpenEditModal}
                setSpace={setSpace}
                isOwner={isOwner}
              />
            )
          })}
        </Grid>
      ) : (
        <Typography variant="body2">{t('SPACES_no_entries')}</Typography>
      )}
      <Fab
        color="secondary"
        aria-label="add"
        variant="extended"
        className="mt-8"
        onClick={() => {
          setOpenModal(true)
        }}
      >
        <AddIcon className="mr-2" /> {t('SPACES_add_space')}
      </Fab>

      <Modal
        open={openModal}
        onClose={() => {
          setOpenModal(false)
        }}
        aria-labelledby="addspace-title"
        aria-describedby="addspace-description"
      >
        <CondensedContainer className="absolute m-0 h-full max-h-full w-full overflow-auto bg-white p-8 drop-shadow-2xl md:top-1/3 md:left-1/2 md:h-[unset] md:w-[600px] md:-translate-x-1/2 md:-translate-y-1/3">
          <Box className="sticky top-0 z-10 flex h-0 w-full justify-end">
            <IconButton
              title={t('BUTTON_close')}
              className="z-10 -mt-2 h-12 w-12 border border-slate-200 bg-white shadow hover:bg-slate-200"
              onClick={() => setOpenModal(false)}
            >
              <CloseIcon />
            </IconButton>
          </Box>
          <h3 id="addspace-title" className="my-0">{`${t(
            'SPACES_add_space'
          )}`}</h3>
          <p id="addspace-description">{t('SPACES_add_space_text')}</p>
          {setOpenModal && (
            <SpaceForm
              isLoading={isLoading}
              setIsLoading={setIsLoading}
              mySpaces={mySpaces}
              setMySpaces={setMySpaces}
              user={user}
              setOpenModal={setOpenModal}
            />
          )}
        </CondensedContainer>
      </Modal>

      <EditSpaceModal
        setAlert={setAlert}
        space={space}
        setSpace={setSpace}
        mySpaces={mySpaces}
        setMySpaces={setMySpaces}
        openEditModal={openEditModal}
        setOpenEditModal={setOpenEditModal}
        t={t}
      />
    </CondensedContainer>
  )
}
