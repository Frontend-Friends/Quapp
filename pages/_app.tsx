import * as React from 'react'
import Head from 'next/head'
import { AppProps } from 'next/app'
import { ThemeProvider } from '@mui/material/styles'
import CssBaseline from '@mui/material/CssBaseline'
import { CacheProvider, EmotionCache } from '@emotion/react'
import theme from '../config/theme'
import createEmotionCache from '../config/create-emotion-cache'
import '../styles/globals.scss'
import { useTranslation } from '../hooks/use-translation'
import Footer from '../components/footer'
import NavigationBar from '../components/navigation/navigation-bar'
import { Notifications } from '../components/notifications'
import { useRouter } from 'next/router'
import { useMemo } from 'react'
import clsx from 'clsx'
import { Snackbar } from '../components/snackbar/snackbar'

const domain = 'quapp.org'
const protocolSubdomain = 'https://www.'

// Client-side cache, shared for the whole session of the user in the browser.
const clientSideEmotionCache = createEmotionCache()

interface MyAppProps extends AppProps {
  emotionCache?: EmotionCache
}

export default function App({
  Component,
  emotionCache = clientSideEmotionCache,
  pageProps,
}: MyAppProps) {
  const t = useTranslation()
  const { asPath } = useRouter()

  const showNavbar = useMemo(() => {
    return asPath != '/'
  }, [asPath])

  return (
    <CacheProvider value={emotionCache}>
      <Head>
        <meta name="viewport" content="initial-scale=1, width=device-width" />
        <meta property="og:title" content={t('HTML_TITLE_general')} />
        <meta property="og:description" content={t('HTML_META_description')} />
        <meta name="description" content={t('HTML_META_description')} />
        <meta
          property="og:image"
          content={`${protocolSubdomain + domain}/og_screenshot.jpg`}
        />
        <meta property="og:type" content="website" />
        <meta property="og:url" content={protocolSubdomain + domain} />
        <meta name="twitter:card" content="summary_large_image" />
        <meta property="twitter:domain" content={domain} />
        <meta property="twitter:url" content={protocolSubdomain + domain} />
        <meta name="twitter:title" content={t('HTML_TITLE_general')} />
        <meta name="twitter:description" content={t('HTML_META_description')} />
        <meta
          name="twitter:image"
          content={`${protocolSubdomain + domain}/og_screenshot.jpg`}
        />

        <title>{t('HTML_TITLE_general')}</title>
      </Head>
      <ThemeProvider theme={theme}>
        <CssBaseline />
        <div
          className={clsx(
            'flex min-h-screen min-w-[340px] flex-col justify-between',
            showNavbar && 'md:pt-[83px]'
          )}
        >
          <div className="grow bg-[url(/assets/img/neighbourhood.svg)] bg-[length:auto_300px] bg-bottom bg-repeat-x">
            <Component {...pageProps} />
          </div>
          <Footer />
        </div>
        {showNavbar && <NavigationBar />}
        <Notifications />
        <Snackbar />
      </ThemeProvider>
    </CacheProvider>
  )
}
