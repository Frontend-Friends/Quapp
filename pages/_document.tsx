import Document, {
  DocumentContext,
  DocumentInitialProps,
  Html,
  Main,
  Head,
  NextScript,
} from 'next/document'
import * as React from 'react'
import createEmotionCache from '../config/create-emotion-cache'
import createEmotionServer from '@emotion/server/create-instance'
import { ReactNode } from 'react'

class MyDocument extends Document<{ emotionStyleTags: ReactNode[] }> {
  static async getInitialProps(
    ctx: DocumentContext
  ): Promise<DocumentInitialProps> {
    return Document.getInitialProps(ctx)
  }

  render() {
    return (
      <Html>
        <Head>
          <link rel="preconnect" href="https://fonts.googleapis.com" />
          <link rel="preconnect" href="https://fonts.gstatic.com" crossOrigin />
          <link
            href="https://fonts.googleapis.com/css2?family=Jost:wght@300;400;500;700&display=swap"
            rel="stylesheet"
          />
          {this.props.emotionStyleTags}
        </Head>
        <body>
          <Main />
          <NextScript />
        </body>
      </Html>
    )
  }
}

MyDocument.getInitialProps = async (ctx) => {
  const originalRenderPage = ctx.renderPage

  const cache = createEmotionCache()
  const { extractCriticalToChunks } = createEmotionServer(cache)

  ctx.renderPage = () =>
    originalRenderPage({
      enhanceApp: (App) =>
        function EnhanceApp(props) {
          // @ts-ignore
          return <App emotionCache={cache} {...props} />
        },
    })

  const initialProps = await Document.getInitialProps(ctx)
  const emotionStyles = extractCriticalToChunks(initialProps.html)
  const emotionStyleTags = emotionStyles.styles.map((style) => (
    <style
      data-emotion={`${style.key} ${style.ids.join(' ')}`}
      key={style.key}
      dangerouslySetInnerHTML={{ __html: style.css }}
    />
  ))

  return {
    ...initialProps,
    emotionStyleTags,
  }
}

export default MyDocument
