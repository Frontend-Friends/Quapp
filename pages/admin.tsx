import { withIronSessionSsr } from 'iron-session/next'

export const getServerSideProps = withIronSessionSsr(
  async function getServerSideProps({ req }) {
    const user = req.session

    if (!user.admin) {
      return {
        notFound: true,
      }
    }

    return {
      props: {
        user: req.session.user,
      },
    }
  },
  {
    cookieName: '__session',
    password: process.env.SECRET_COOKIE_PASSWORD as string,
    // secure: true should be used in production (HTTPS) but can't be used in development (HTTP)
    cookieOptions: {
      secure: process.env.NODE_ENV === 'production',
    },
  }
)
