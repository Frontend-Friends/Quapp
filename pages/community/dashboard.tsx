import { FC, useState } from 'react'
import { CondensedContainer } from '../../components/condensed-container'
import { useTranslation } from '../../hooks/use-translation'
import { Fab, Grid, Typography } from '@mui/material'
import { Header } from '../../components/header'
import AddIcon from '@mui/icons-material/Add'
import { InferGetServerSidePropsType } from 'next'
import { SpaceItemType } from '../../components/products/types'
import { getDoc } from 'firebase/firestore'
import SpaceItem from '../../components/spaces/space-item'
import { withIronSessionSsr } from 'iron-session/next'
import { sessionOptions } from '../../config/session-config'
import { getSpaceRef } from '../../lib/helpers/refs/get-space-ref'
import SpaceForm from './space-form'
import { fetchUser } from '../../lib/services/fetch-user'

export const getServerSideProps = withIronSessionSsr<{
  spaces?: SpaceItemType[]
}>(async ({ req }) => {
  const { user } = req.session
  if (!user) {
    return { props: {} }
  }
  const fetchedUser = await fetchUser(user.id ?? '')
  const spaces = await Promise.all<SpaceItemType>(
    fetchedUser.spaces?.map(
      (space) =>
        new Promise(async (resolve) => {
          const [ref] = getSpaceRef(space)
          const fetchedDoc = await getDoc(ref).then((result) => {
            const data = result.data()
            return {
              ...data,
              id: result.id,
              ownerId: data?.ownerId.id || '',
              creatorId: data?.creatorId.id || '',
              creationDate: data?.creationDate?.seconds ?? 0,
            } as SpaceItemType
          })
          resolve(fetchedDoc)
        })
    ) || []
  )
  return { props: { spaces } }
}, sessionOptions)
const Dashboard: FC<InferGetServerSidePropsType<typeof getServerSideProps>> = ({
  spaces,
}) => {
  const t = useTranslation()
  const [open, setOpen] = useState(false)
  const [isLoading, setIsLoading] = useState(false)
  const [message, setMessage] = useState('')

  return (
    <CondensedContainer className="relative">
      <Header title={t('SPACES_title')} />
      {!!spaces?.length ? (
        <Grid container columns={{ md: 1 }} spacing={{ xs: 4 }} pt={4}>
          {spaces.map((space) => (
            <SpaceItem key={space.id} space={space} />
          ))}
        </Grid>
      ) : (
        <Typography variant="body2">{t('SPACES_no_entries')}</Typography>
      )}
      <Fab
        color="primary"
        aria-label="add"
        variant="extended"
        className="mt-8"
        onClick={() => setOpen((state) => !state)}
      >
        <AddIcon className="mr-2" /> {t('SPACES_add_space')}
      </Fab>
      {open && (
        <SpaceForm
          setOpen={setOpen}
          open={open}
          isLoading={isLoading}
          setIsLoading={setIsLoading}
          setMessage={setMessage}
          message={message}
        />
      )}
    </CondensedContainer>
  )
}

export default Dashboard
