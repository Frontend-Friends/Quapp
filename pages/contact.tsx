import { CondensedContainer } from '../components/condensed-container'

const Contact = () => {
  return (
    <CondensedContainer>
      <h1>Kontaktdaten</h1>
      <p>Laurent Kerbage, Philip Park, Fabrice Tobler</p>
      <p>
        E-Mail: quapp.shared@gmail.com
        <br />
        WebSite: https://www.quapp.org/
      </p>
    </CondensedContainer>
  )
}
export default Contact
