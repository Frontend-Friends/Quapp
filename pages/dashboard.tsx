import React from 'react'
import { withIronSessionSsr } from 'iron-session/next'
import { ironOptions } from '../lib/config'
import { User } from '../interfaces/user'

export const getServerSideProps = withIronSessionSsr(async ({ req }) => {
  const { user } = req.session
  return {
    props: { isLoggedIn: !!user, user },
  }
}, ironOptions)

const Dashboard: React.FC<{ user?: User }> = ({ user }) => {
  return (
    <div>
      <h1> Hello {user?.firstName}!</h1> This route is protected. You can reach
      it by applying correct credentials.
    </div>
  )
}

export default Dashboard
