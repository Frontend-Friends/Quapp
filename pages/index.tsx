import { withIronSessionSsr } from 'iron-session/next'
import { ironOptions } from '../lib/config'
import { FC, useEffect, useState } from 'react'
import { LogoSVG } from '../components/svg/quapp_logo'
import { AppartmentSVG } from '../components/svg/appartment'
import { WindowNeighboursSVG } from '../components/svg/windowneighbours'
import { CollaboratorSVG } from '../components/svg/collaborator'
import { HangoutSVG } from '../components/svg/hangout'
import { Box, Button, Divider, Link } from '@mui/material'
import { useTranslation } from '../hooks/use-translation'
import { WaveWhiteSVG } from '../components/svg/wave_white'
import { WaveWhiteBottomSVG } from '../components/svg/wave_white_bottom'
import { WaveWhiteTopSVG } from '../components/svg/wave_white_top'
import { useRouter } from 'next/router'
import { fetchJson } from '../lib/helpers/fetch-json'

export const getServerSideProps = withIronSessionSsr(async ({ req }) => {
  const { user } = req.session
  return {
    props: { isLoggedIn: !!user },
  }
}, ironOptions)

const userExists = async () => {
  const user: { isUser: boolean } = await fetchJson(' /api/cookie')
  return user.isUser
}

const Home: FC<{ isLoggedIn: boolean }> = () => {
  const t = useTranslation()
  const [isUser, setIsUser] = useState(false)
  const { asPath } = useRouter()

  useEffect(() => {
    const user = userExists().then((res) => res)
    user.then((res) => setIsUser(res))
  }, [asPath])

  return (
    <main className="w-full text-blueishGray-600">
      <section className="relative bg-gradient-to-b from-violetRed-600 via-violetRed-600 to-violetRed-900 pb-10 text-white">
        <div className="mx-auto grid max-w-[1280px] grid-cols-7 grid-rows-[auto_auto_auto_auto_auto_auto_auto] p-3 lg:p-8 lg:pb-24">
          <LogoSVG
            aria-label={t('SVG_logo')}
            className="col-span-7 mx-auto block w-1/2 md:h-full md:w-1/5 lg:mb-10"
            preserveAspectRatio="xMidYMid meet"
          />
          <h1 className="col-span-7 text-center text-xl font-medium sm:col-start-3 sm:text-left md:text-xl lg:pl-14 lg:text-2xl 2xl:col-start-4">
            {t('HOME_subtitle')}
          </h1>
          <AppartmentSVG
            aria-label={t('SVG_appartment')}
            className="col-span-3 row-span-2 -ml-6 h-[130px] w-full flex-1 sm:col-span-2 sm:-ml-14 sm:scale-125 md:scale-150 lg:scale-[2] 2xl:col-span-3 2xl:-ml-0 2xl:scale-[2.3]"
          />
          <p className="col-span-4 row-span-2 m-0 flex-1 text-lg sm:col-span-4 md:text-xl lg:row-span-1 lg:pl-14 2xl:col-span-4">
            {!isUser ? t('HOME_intro') : t('HOME_is_logged_in')}
          </p>

          {!isUser ? (
            <>
              <Link
                underline="none"
                href="/auth/signup"
                className="col-span-7 mx-auto mt-6 mb-5 block md:col-start-3 md:mx-[unset] md:mt-0 lg:mb-6 lg:mt-8 lg:pl-14 2xl:col-span-4"
              >
                <Button
                  type="submit"
                  variant="contained"
                  color="secondary"
                  className="px-12 py-3 md:text-lg lg:py-4 lg:text-xl"
                >
                  {t('HOME_signup_free')}
                </Button>
              </Link>
              <Box className="col-span-7 text-center md:col-start-3 md:text-left lg:pl-14 lg:text-xl 2xl:col-start-4">
                <Link
                  underline="hover"
                  href="/auth/login"
                  className="text-white underline"
                >
                  {t('LOGIN_has_account')}
                </Link>
              </Box>
            </>
          ) : (
            <>
              <Link
                underline="none"
                href="/community/dashboard"
                className="col-span-7 mx-auto mt-6 mb-5 block md:col-start-3 md:mx-[unset] md:mt-0 lg:mb-6 lg:mt-8 lg:pl-14 2xl:col-span-4"
              >
                <Button
                  type="submit"
                  variant="contained"
                  color="secondary"
                  className="px-12 py-3 md:text-lg lg:py-4 lg:text-xl"
                >
                  {t('HOME_to_dashboard')}
                </Button>
              </Link>
            </>
          )}
        </div>
        <WaveWhiteSVG
          preserveAspectRatio="none"
          className="xl:h-30 pointer-events-none absolute bottom-0 w-full touch-none sm:h-12 md:h-14 lg:h-28"
        />
      </section>
      <section className="mx-auto max-w-[1280px] flex-row items-center p-3 lg:flex lg:p-8 lg:pt-14">
        <WindowNeighboursSVG
          aria-label={t('SVG_windowneighbours')}
          className="w-full lg:w-1/2"
        />
        <div className="sm:p-3 md:pt-12 lg:order-first lg:w-1/2">
          <h2 className="m-0 text-center font-medium md:text-2xl lg:text-left lg:text-3xl">
            {t('HOME_support_title')}
          </h2>
          <Divider className="my-3 lg:my-6" />
          <p className="mt-1 text-center text-lg md:text-xl lg:pr-6 lg:text-left">
            {t('HOME_support_text')}
          </p>
        </div>
      </section>
      <section className="mb-10 pb-10 lg:p-8">
        <div className="relative mx-auto max-w-[1280px] bg-gradient-to-t from-slate-300 to-slate-200">
          <WaveWhiteTopSVG
            preserveAspectRatio="none"
            className="absolute top-0 w-full sm:h-24 lg:h-28"
          />
          <div className="pb-32 text-center sm:pb-20 md:pb-20 lg:flex lg:items-center lg:py-20">
            <CollaboratorSVG
              aria-label={t('SVG_collaborator')}
              className="relative h-[256px] lg:basis-2/6"
            />
            <div className="relative z-10 p-3 lg:flex lg:basis-4/6 lg:gap-10 lg:p-8">
              <div className="text-left sm:p-3">
                <h2 className="m-0 text-2xl font-medium md:text-3xl">
                  {t('HOME_spaces_title')}
                </h2>
                <Divider className="my-3 w-1/2 lg:my-6" />
                <p className="mt-1 text-lg md:text-xl lg:text-xl">
                  {t('HOME_spaces_text')}
                </p>
              </div>
              <div className="text-right sm:p-3 lg:text-left">
                <h2 className="m-0 text-2xl font-medium md:text-3xl">
                  {t('HOME_offers_title')}
                </h2>
                <Divider className="my-3 ml-auto w-1/2 lg:my-6 lg:ml-[unset]" />
                <p className="mt-1 text-lg md:text-xl lg:text-xl">
                  {t('HOME_offers_text')}
                </p>
              </div>
            </div>
          </div>
          <WaveWhiteBottomSVG
            preserveAspectRatio="none"
            className="absolute bottom-0 w-full sm:h-24 lg:h-28"
          />
        </div>
      </section>
      <section className="relative mx-auto flex max-w-[1280px] flex-col items-center bg-white p-3 pb-16 lg:flex-row lg:flex-wrap lg:p-8">
        <h2 className="absolute left-1/3 z-10 m-0 w-1/2 text-left text-4xl font-medium lg:static lg:order-1 lg:flex-[0_0_100%]">
          {t('HOME_support_title')}
        </h2>
        <HangoutSVG
          aria-label={t('SVG_hangout')}
          className="-ml-20 w-full lg:order-4 lg:ml-auto lg:h-[384px] lg:flex-[0_0_33%] lg:shrink lg:scale-x-[-1]"
        />
        <Divider className="mt-7 w-full lg:order-2" />
        <p className="text-center text-lg md:text-xl lg:order-3 lg:flex-[0_0_66%] lg:self-start lg:text-left lg:text-2xl">
          {t('HOME_support_text')}
        </p>
      </section>
    </main>
  )
}
export default Home
