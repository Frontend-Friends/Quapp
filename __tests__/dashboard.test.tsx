import { render } from '@testing-library/react'
import { Dashboard } from '../components/pages/dashboard'

it('renders Dashboard unchanged', () => {
  const { container } = render(<Dashboard t={(key) => key} />)
  expect(container).toMatchSnapshot()
})
