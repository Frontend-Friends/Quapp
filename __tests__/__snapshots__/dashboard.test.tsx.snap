// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Dashboard unchanged 1`] = `
<div>
  <div
    class="mx-auto my-[10%] w-full max-w-3xl p-2 md:p-8 relative MuiBox-root css-0"
  >
    <header>
      <h1
        class="MuiTypography-root MuiTypography-h1 text-violetRed-600 md:text-3xl lg:text-4xl css-o2w69a-MuiTypography-root"
      >
        SPACES_title
      </h1>
    </header>
    <p
      class="MuiTypography-root MuiTypography-body2 css-e784if-MuiTypography-root"
    >
      SPACES_no_entries
    </p>
    <button
      aria-label="add"
      class="MuiButtonBase-root MuiFab-root MuiFab-extended MuiFab-sizeLarge MuiFab-primary MuiFab-root MuiFab-extended MuiFab-sizeLarge MuiFab-primary mt-8 css-nubenn-MuiButtonBase-root-MuiFab-root"
      tabindex="0"
      type="button"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium mr-2 css-i4bv87-MuiSvgIcon-root"
        data-testid="AddIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M19 13h-6v6h-2v-6H5v-2h6V5h2v6h6v2z"
        />
      </svg>
       
      SPACES_add_space
      <span
        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
      />
    </button>
  </div>
</div>
`;
